### PR TITLE
Change date and datetime handling in the Go SDK

### PR DESCRIPTION
**Pull request by ronanwatkins**
Required to facilitate handling for 3 different date formats:
* yyyy-MM-ddTHH:mm:ss.SSS
* yyyy-MM-ddTHH:mm:ss.SSSZ
* yyyy-MM-dd

Related: https://github.com/purecloudlabs/swagger-codegen/pull/39

### CODE REVIEW

File: resources/sdk/purecloudgo/templates/model.mustache [hunk 1]
```diff
@@ -15,6 +16,45 @@ type {{classname}} struct { {{#vars}}
...
 
+func (u *{{classname}}) MarshalJSON() ([]byte, error) {
+	// Redundant initialization to avoid unused import errors for models with no Time values
+	_  = timeutil.Timedelta{}
+	type Alias {{classname}}
+
+	{{#vars}}{{#isDateTime}}
+	{{name}} := new(string)
+	if u.{{name}} != nil {
+		{{#vendorExtensions.x-local-date-time}}*{{name}} = timeutil.Strftime(u.{{name}}, "%Y-%m-%dT%H:%M:%S.%f"){{/vendorExtensions.x-local-date-time}}
```
Comment (timsmithgenesys) [inline resources/sdk/purecloudgo/templates/model.mustache:27]: Are these valid if there's no milliseconds? There are 4 known formats used by the Java SDK: https://github.com/purecloudlabs/platform-client-sdk-common/blob/master/resources/sdk/purecloudjava/extensions/ApiDateFormat.java#L14.

Comment (timsmithgenesys) [reply resources/sdk/purecloudgo/templates/model.mustache:27]: Any of them. I believe it's an API-layer behavior that truncates the milliseconds when they're exactly zero.

Comment (ronanwatkins) [reply resources/sdk/purecloudgo/templates/model.mustache:27]: That should be ok then. This change was more for formatting times to be sent to the API as a request payload.
I doubt there would be any issues arising from a client expecting `2021-08-04T08:00:00Z`  and receiving `2021-08-04T08:00:00.00Z` instead.

Comment (timsmithgenesys) [reply resources/sdk/purecloudgo/templates/model.mustache:27]: AFAIK that's accurate. I believe the API is tolerant of optional ms.

**Pull request merged**